### PR TITLE
Define TS types for the useCollectionData() hook

### PR DESCRIPTION
**Pull request by retrofox**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This is a janitorial PR follow-up of https://github.com/woocommerce/woocommerce/pull/52458.

We consider the `useCollectionData()` may not use the useCollection() hook since it should always return a collection (Array) while `useCollectionData()` returns an object.
In the meantime, we can tidy the code to address a few TS issues/enhancements:

* Define the TS types of the useCollectionData() statically. [feedback](https://github.com/woocommerce/woocommerce/pull/52458#pullrequestreview-2409381630) 
* Remove the type `< T >` parameter from the function definition 
* Update and tweak the types usage/definition to address TS issues in respective files

Closes # .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

This PR doesn't change the functionality of the app since all are TypeScript changes. Thus:

* Confirm the changes does make sense
* Confirm what the hook returns has the expected type; for instance, in the image below, the IDE properly identifies the data structure of the [filteredCounts](https://github.com/woocommerce/woocommerce/blob/9c57fa043f118f197b8d9b2e023c562daf63b684/plugins/woocommerce-blocks/assets/js/blocks/attribute-filter/block.tsx#L142) const:

image

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [x] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts [hunk 1]
```diff
@@ -229,25 +229,27 @@ type StockStatusCount = {
...
+	min_price: string;
+	max_price: string;
+	currency_code: CurrencyCode;
+	currency_decimal_separator: '.' | string;
+	currency_minor_unit: number;
+	currency_prefix: '$' | string;
+	currency_suffix: '' | string;
+	currency_symbol: '$' | string;
+	currency_thousand_separator: ',' | string;
+};
```
Comment (dinhtungdu) [inline plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts]:
```ts
type PriceRangeProps = CurrencyResponse & {
	min_price: string;
	max_price: string;
};
```

What do you think about composing types like this?

File: plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts [hunk 2]
```diff
@@ -229,25 +229,27 @@ type StockStatusCount = {
...
+
 /*
  * Prop types for the `wc/store/v1/products/collection-data` endpoint
  */
 export type WCStoreV1ProductsCollectionProps = {
```
Comment (dinhtungdu) [inline plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts]: This is out of the scope of this PR but I don't think the Product Collection block is a good place to export this type. This is the return type of a hook, so `woocommerce/plugins/woocommerce-blocks/assets/js/types/type-defs/hooks.ts` sounds more ideal to me.

File: plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection-data.ts [hunk 3]
```diff
@@ -155,7 +156,7 @@ export const useCollectionData = < T >( {
 		return buildCollectionDataQuery( collectionDataQueryState );
 	}, [ collectionDataQueryState ] );
 
-	return useCollection< T >( {
+	return useCollection< WCStoreV1ProductsCollectionProps, false >( {
```
Comment (dinhtungdu) [inline plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection-data.ts]: `false` is the default value for the second parameter, so what is the benefit of passing it? For explicitly?

This is also out of the scope of this PR, but I want to start the conversation about the type of `useCollection`:
- Its current type makes its usage inconsistent. If you want to return an array, you have two ways:
  1. Passing the type of array item for the first argument and `true` for the second. We don't do this anywhere atm.
  2. Passing the array type as the first argument, like we're doing in https://github.com/woocommerce/woocommerce/blob/8ab7594801a5aba1eccb443f108f4247e765bae9/plugins/woocommerce-blocks/assets/js/base/context/hooks/use-store-products.ts#L38-L43
- Because the only place the `useCollection` hook doesn't return the array is inside `useCollectionData`, I think we should restore the original type of `useCollection` and cast its return type in `useCollectionData`.

**Review by dinhtungdu — COMMENTED** [review]
Thanks for working on this! The PR looks good in general. I have a minor comment about composing the type. Apart from that, the other comments are just conversation stater and should be discussed further, feel free to ignore them for this PR.

File: plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts (hunk 1 above)
Comment (retrofox) [reply plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts]: Let's do it 👍

File: plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection-data.ts (hunk 3 above)
Comment (retrofox) [reply plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection-data.ts]: > `false` is the default value for the second parameter, so what is the benefit of passing it? For explicitly?

That's correct. I guess we can get rid of it now.

> * Because the only place the `useCollection` hook doesn't return the array is inside `useCollectionData`, I think we should restore the original type of `useCollection` and cast its return type in `useCollectionData`.

After diving into the code, you're right. I agree the best solution is restoring the type handling in the useCollection hook.

File: plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts (hunk 1 above)
Comment (retrofox) [reply plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts]: what about predefined types

For instance:

```ts
currency_prefix: '$' | string;
```

It defines the `$`? Could we move that in the `CurrencyResponse` type def?

File: plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts (hunk 2 above)
Comment (retrofox) [reply plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts]: let's move it on this PR

File: plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts (hunk 1 above)
Comment (dinhtungdu) [reply plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts]: Can you explain why do we need a default value for the prefix?

Comment (dinhtungdu) [reply plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts]: And other properties, too. It's not clear to me why we need them.

Comment (retrofox) [reply plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts]: To me, default values offer convenience and clarity, especially in an IDE. They help establish a standard setup (like $ for currency prefix) while allowing flexibility when needed. IDEs use these defaults to suggest expected values.
Finally, it does not hurt

Comment (retrofox) [reply plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts]: Updated PR (sorry).
For instance, in the image below, when we compare the `currency_simbol`, the IDE shows the default values. It doesn't mean that is not possible to use a regular string, tho.

image

Comment (dinhtungdu) [reply plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts]: > It doesn't mean that is not possible to use a regular string, tho.

From your screenshot, it gives me the impression that the property is a union type with 5 available values.

Comment (retrofox) [reply plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts]: But it isn't. It allows to use of any string:

image

No issues above.  Also, it's define in the type

Comment (retrofox) [reply plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts]: picked more symbols from [here](https://github.com/woocommerce/woocommerce/blob/e62c541d8f3adb8568bc33fd30c729185dbdff7f/plugins/woocommerce/i18n/locale-info.php)

Comment (dinhtungdu) [reply plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts]: But isn't it against the original purpose — offering convenience and clarity? 

For example, from the suggestion, I think it's a union type. But then I figured out: Oh, it's not; it's a string. It seems to me that we're losing clarity here. Is there any issue with my thought process? Is my assumption on the union type correct? Happy to change my mind if I interpret it wrong.

Comment (dinhtungdu) [reply plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts]: Also, maybe we should loop more people in for more diverse opinions?

File: plugins/woocommerce-blocks/assets/js/types/type-defs/currency.ts [hunk 4]
```diff
@@ -33,14 +33,87 @@ export interface Currency {
...
 	currency_code: CurrencyCode;
-	currency_symbol: string;
+
+	currency_symbol:
+		| CurrencyPrefix
+		| CurrencySuffix
+		| CurrencyBoth
+		| ( string & {} ); // eslint-disable-line @typescript-eslint/ban-types
+
+	currency_prefix: CurrencyPrefix | CurrencyBoth;
```
Comment (kmanijak) [inline plugins/woocommerce-blocks/assets/js/types/type-defs/currency.ts]: I guess it can be empty string the same as suffix 🤔 

```suggestion
	currency_prefix: '' | CurrencyPrefix | CurrencyBoth;
```

File: plugins/woocommerce-blocks/assets/js/types/type-defs/hooks.ts [hunk 5]
```diff
@@ -69,3 +69,38 @@ export type Query = {
 	orderby: string;
 	order: string;
 };
+
+export type RatingValues = 0 | 1 | 2 | 3 | 4 | 5;
+
+type AttributeCount = {
```
Comment (kmanijak) [inline plugins/woocommerce-blocks/assets/js/types/type-defs/hooks.ts]: Nitpick: it can be used in Filter by Attribute:
https://github.com/woocommerce/woocommerce/blob/30cc92fddc42277a3e7d1ce26e5a044e72265980/plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/attribute-filter/types.ts#L27

**Review by kmanijak — COMMENTED** [review]
Thanks for this PR! I left some nitpick comments. But I'm not approving due to ongoing discussion re currency symbols.

File: plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts (hunk 1 above)
Comment (kmanijak) [reply plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts]: I'm not convinced about listing the currency symbols for two reasons:

-  list is incomplete (found at least 2 missing currencies supported by WordPress: `New Taiwan Dollar NT$ - NTD` or `Yemeni rial ﷼ - YER` but there's more).
- list doesn't support three letter codes (e.g. EUR, PLN etc.) Some currencies don't have symbols at all and only use code ISO 4217 codes.

So to sum up I'm either up for full list (which I don't think we really need, it doesn't bring much value compared to effort) or keeping it `string` as it was 😅

Comment (retrofox) [reply plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts]: > But isn't it against the original purpose — offering convenience and clarity?

I don't think so. It keeps offering convenience and clarity and now also suggests valid typical values.

> For example, from the suggestion, I think it's a union type.

It is!

> But then I figured out: Oh, it's not; it's a string.

It's a union type: `literals[] ` + `string`.

> It seems to me that we're losing clarity here. Is there any issue with my thought process? Is my assumption on the union type correct? Happy to change my mind if I interpret it wrong.

To me, it's pretty clear and straightforward. The type has valid common types but also has the flexibility to define any other values if needed.

Comment (retrofox) [reply plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts]: > Also, maybe we should loop more people in for more diverse opinions?

Yes, or maybe I should roll back these changes to move on.

File: plugins/woocommerce-blocks/assets/js/types/type-defs/hooks.ts (hunk 5 above)
Comment (retrofox) [reply plugins/woocommerce-blocks/assets/js/types/type-defs/hooks.ts]: nice one. addressed here cb95de78a2fb6aa167c15a89f049e28511e53137

File: plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts (hunk 1 above)
Comment (retrofox) [reply plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts]: > I'm not convinced about listing the currency symbols for two reasons:
> 
> * list is incomplete (found at least 2 missing currencies supported by WordPress: `New Taiwan Dollar NT$ - NTD` or `Yemeni rial ﷼ - YER` but there's more).

The list was composed by picking the symbols from here [woocommerce/i18n/locale-info.php](https://github.com/woocommerce/woocommerce/blob/e62c541d8f3adb8568bc33fd30c729185dbdff7f/plugins/woocommerce/i18n/locale-info.php). 

Also, that's the point of this approach: although the list is not complete, something that we can work on is adding other code, symbols, etc., it's possible to use any `string` value.

> * list doesn't support three letter codes (e.g. EUR, PLN etc.) Some currencies don't have symbols at all and only use code ISO 4217 codes.

It's just a list of literal values. You can use whatever you need since, in the end, it also accepts strings.
 
> So to sum up I'm either up for full list (which I don't think we really need, it doesn't bring much value compared to effort) or keeping it `string` as it was 😅

There is no need to keep the list updated. All currency properties also accept random generica values.

Comment (dinhtungdu) [reply plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts]: I can see the benefit of having the editor suggest values, but I'm still not convinced. In VSCode, I get this when trying to check the type of `currency_symbol` property:

image

I can't figure out the exact type of `currency_symbol` property by hovering on it. I can't click on the more button in the middle to expand the list, too.

By the way, does it have `undefined` type?

I tried reducing the suggestion list to 6 values, and here is what I got

image

To me, it's a very complex type that I can't understand without going to the type definition and some effort (to understand `string & {}` type).

In summary, the benefit of the complex type here doesn't outperform the complexity it adds to me. So I'm not convinced.

File: plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection.ts [hunk 6]
```diff
@@ -131,7 +129,5 @@ export const useCollection = < T, IsCollection extends boolean = false >(
 		currentResults.current = results;
 	}
 
-	return currentResults.current as IsCollection extends true
-		? { results: T[]; isLoading: boolean }
-		: { results: T; isLoading: boolean };
+	return currentResults.current as { results: T; isLoading: boolean };
```
Comment (dinhtungdu) [inline plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection.ts]: We don't need typecasting here anymore. Also, we will need to restore the comment of this hook. I suggest reverting the whole file to this point: https://github.com/woocommerce/woocommerce/blob/ebb300c1e7288b869cba259feb6279bc6cc6fc8e/plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection.ts for simplicity.

Comment (dinhtungdu) [reply plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection.ts]: We also need to revert the change we made for callers of this hook too. Like one in `useStoreProducts`.

**Review by dinhtungdu — CHANGES_REQUESTED** [review]
Thanks for the update! I missed reviewing the `use-collection.ts` file so here is one more round. I'm sorry about incremental reviews.

File: plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection.ts (hunk 6 above)
Comment (retrofox) [reply plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection.ts]: > We don't need typecasting here anymore. Also, we will need to restore the comment of this hook. I suggest reverting the whole file to this point:

It's not possible. If we do, we'll again introduce hundreds of TS errors using the useCollectionData() hook.

Whether we like it or not, the current code allows useCollection() to return not only a [collection of items](https://github.com/woocommerce/woocommerce/blob/ebb300c1e7288b869cba259feb6279bc6cc6fc8e/plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection.ts#L55) but also an object. [Here](https://github.com/woocommerce/woocommerce/blob/ebb300c1e7288b869cba259feb6279bc6cc6fc8e/plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection-data.ts#L158), the useCollectionData gets an object.

The underlying problem is, again, that the useCollection() hook sometimes does not return a collection, but an object.

Comment (retrofox) [reply plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection.ts]: > We also need to revert the change we made for callers of this hook too. Like one in `useStoreProducts`.

We need to do this if we remove the typecasting.
The final solution is to decouple the useCollection() from the userCollectionData(). imo.

**Review by dinhtungdu — COMMENTED** [review]
> It's not possible. If we do, we'll again introduce hundreds of TS errors using the useCollectionData() hook.

Instead of casting the type in the `useCollection`, we can do it in `useCollectionData` instead. Here is what I mean: https://github.com/woocommerce/woocommerce/compare/update/define-use-collection-data-types...try/only-casting-in-useCollectionData.

> Whether we like it or not, the current code allows useCollection() to return not only a [collection of items](https://github.com/woocommerce/woocommerce/blob/ebb300c1e7288b869cba259feb6279bc6cc6fc8e/plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection.ts#L55) but also an object. [Here](https://github.com/woocommerce/woocommerce/blob/ebb300c1e7288b869cba259feb6279bc6cc6fc8e/plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection-data.ts#L158), the useCollectionData gets an object.

Yeah, you're right.

> The final solution is to decouple the useCollection() from the userCollectionData(). imo.

It's still possible to use `useCollection` with any route that may or may not return a collection. So we will need to do something more, such as limiting the routes it can use. But that's for the future. For now, the only problem I have with the current type is, even it's technically correct, that `useCollection` can return non-collection data, it's just confusing to me.

The reason I suggest moving the type casting to `useCollectionData` is:
- to avoid updating the usages of `useCollection` in the future (current and new usage).
- `useCollectionData` is the only place that needs a non-collection from `useCollection`.

But, even I don't like it, the current type of `useCollection` reflects its returned data type better than the previous one, so I agree to go with the current and improve on it later.

There are (last) two minor comments:
- Can you please fix the comment [here](https://github.com/woocommerce/woocommerce/blob/5d2b24bb8eafc8cd3a67a99aba060ad63eb1f00f/plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection.ts#L40-L41)? It's an array or object depending on the data returned from the endpoint, not because of the number of item in the collection.
- As the hook now takes T and returns T, I don't think we need to use array [here](https://github.com/woocommerce/woocommerce/blob/5d2b24bb8eafc8cd3a67a99aba060ad63eb1f00f/plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection.ts#L73), [here](https://github.com/woocommerce/woocommerce/blob/5d2b24bb8eafc8cd3a67a99aba060ad63eb1f00f/plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection.ts#L76) and [here](https://github.com/woocommerce/woocommerce/blob/5d2b24bb8eafc8cd3a67a99aba060ad63eb1f00f/plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection.ts#L110). But this PR is taking longer than it should be, feel free to do it in a follow up.

**Comment by retrofox** [comment]
> > It's not possible. If we do, we'll again introduce hundreds of TS errors using the useCollectionData() hook.
> 
> Instead of casting the type in the `useCollection`, we can do it in `useCollectionData` instead. Here is what I mean: [update/define-use-collection-data-types...try/only-casting-in-useCollectionData](https://github.com/woocommerce/woocommerce/compare/update/define-use-collection-data-types...try/only-casting-in-useCollectionData).

👍 It does make sense.

> 
> > Whether we like it or not, the current code allows useCollection() to return not only a [collection of items](https://github.com/woocommerce/woocommerce/blob/ebb300c1e7288b869cba259feb6279bc6cc6fc8e/plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection.ts#L55) but also an object. [Here](https://github.com/woocommerce/woocommerce/blob/ebb300c1e7288b869cba259feb6279bc6cc6fc8e/plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection-data.ts#L158), the useCollectionData gets an object.
> 
> Yeah, you're right.
> 
> > The final solution is to decouple the useCollection() from the userCollectionData(). imo.
> 
> It's still possible to use `useCollection` with any route that may or may not return a collection. So we will need to do something more, such as limiting the routes it can use. But that's for the future. For now, the only problem I have with the current type is, even it's technically correct, that `useCollection` can return non-collection data, it's just confusing to me.

Maybe we could also cast the response server-side or even in the client, ensuring that an array (collection) is returned even when it contains a single object.

> 
> The reason I suggest moving the type casting to `useCollectionData` is:
> 
> * to avoid updating the usages of `useCollection` in the future (current and new usage).
> * `useCollectionData` is the only place that needs a non-collection from `useCollection`.
> 
> But, even I don't like it, the current type of `useCollection` reflects its returned data type better than the previous one, so I agree to go with the current and improve on it later.

👍 

> There are (last) two minor comments:
> 
> * Can you please fix the comment [here](https://github.com/woocommerce/woocommerce/blob/5d2b24bb8eafc8cd3a67a99aba060ad63eb1f00f/plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection.ts#L40-L41)? It's an array or object depending on the data returned from the endpoint, not because of the number of item in the collection.

Sure thing!

> * As the hook now takes T and returns T, I don't think we need to use array [here](https://github.com/woocommerce/woocommerce/blob/5d2b24bb8eafc8cd3a67a99aba060ad63eb1f00f/plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection.ts#L73), [here](https://github.com/woocommerce/woocommerce/blob/5d2b24bb8eafc8cd3a67a99aba060ad63eb1f00f/plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection.ts#L76) and [here](https://github.com/woocommerce/woocommerce/blob/5d2b24bb8eafc8cd3a67a99aba060ad63eb1f00f/plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection.ts#L110). But this PR is taking longer than it should be, feel free to do it in a follow up.

Same. Just FYI, the last change you suggested I think shouldn't be done:

```ts
results: store.getCollection< T[] >( ...args ),
```

**Comment by retrofox** [comment]
HI, @dinhtungdu.
I've addressed all feedback suggested, and also I dared to rename what the hook returns, replacing `results` by `data`.
IMO, `results` is very tied to a collection.
Happy to revert that, tho.

**Review by dinhtungdu — APPROVED** [review]
Thanks for addressing all the feedback! This is LGTM! Great work 🎉 :shipit:

**Review by dinhtungdu — APPROVED** [review]
Thanks for addressing all the feedback! This is LGTM! Great work 🎉 :shipit:

**Comment by dinhtungdu** [comment]
@retrofox I run the test locally and the tests passed at 241ceedcc998951677e8cb1d730329c3b2de034e
image


But with the latest commit (7c358353008ffeedffc35e7f3688902c76194f01), it failed:
image

**Comment by retrofox** [comment]
> @retrofox I run the test locally and the tests passed at [241ceed](https://github.com/woocommerce/woocommerce/commit/241ceedcc998951677e8cb1d730329c3b2de034e) image
> 
> But with the latest commit ([7c35835](https://github.com/woocommerce/woocommerce/commit/7c358353008ffeedffc35e7f3688902c76194f01)), it failed: image

looking...

**Comment by retrofox** [comment]
Addressed failing [tests here](https://github.com/woocommerce/woocommerce/pull/52586/commits/0ce5430c91d219bbbd38ddeba82bb02b3dd04921), @dinhtungdu 🤞
